pkg/kinflate/tree/default.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tree
+
+import (
+	"k8s.io/kubectl/pkg/kinflate/transformers"
+	"k8s.io/kubectl/pkg/kinflate/types"
+)
+
+// DefaultTransformer generates the following transformers:
+// 1) apply overlay
+// 2) name prefix
+// 3) apply labels
+// 4) apply annotations
+// 5) update name reference
+func DefaultTransformer(m *ManifestData) (transformers.Transformer, error) {
+	ts := []transformers.Transformer{}
+
+	ot, err := transformers.NewOverlayTransformer(types.KObject(m.Patches))
+	if err != nil {
+		return nil, err
+	}
+	if ot != nil {
```

Comment:
Shouldn't this be an error ?
Generally, if we didn't return err above, we should not get an empty transformer.

Reply:
same comment below.

Reply:
Will addresses in a followup PR.